Fixes and improvements for building and packaging

dpward opened the pull request:
Supersedes: #55

sgalgano commented:
Package building on CentOS 7, Fedora 23 and Ubuntu 14.04 all pass.

Build is having an issue on CentOS 6.8 due to autoconf version:
```
[emane@localhost emane-0.9.3]$ ./autogen.sh 
configure.ac:132: error: Autoconf version 2.64 or higher is required
m4/ax_check_compile_flag.m4:60: AX_CHECK_COMPILE_FLAG is expanded from...
configure.ac:132: the top level
autom4te: /usr/bin/m4 failed with exit status: 63
aclocal: autom4te failed with exit status: 63
autoreconf: aclocal failed with exit status: 63
```
```
[emane@localhost emane-0.9.3]$ autoconf --version
autoconf (GNU Autoconf) 2.63
Copyright (C) 2008 Free Software Foundation, Inc.
License GPLv2+: GNU GPL version 2 or later
<http://gnu.org/licenses/old-licenses/gpl-2.0.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David J. MacKenzie and Akim Demaille.
```
EPEL makes available autoconf 2.68, but the preference is to build with the base version.

Current support targets are:
* CentOS 6/7 (EL6/EL7)
* Fedora 23/24
* Ubuntu 14.04/16.04

dpward commented:
Thanks for the heads up. I was away and unable to access e-mail this past week. I've updated the pull request and also rebased it on top of your additional commits while I was gone.

Commit 1c68a1f is modified to include the previous version of the `AX_CHECK_COMPILE_FLAG` ([view diff](http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blobdiff;f=m4/ax_check_compile_flag.m4;h=ca3639715e7243fa4343eb2e7f3de53927cc9d4a;hp=51df0c09a7a8bf40dc11193559a45296545653c5;hb=1a869696e4129279f7b99c3f9052717354b79a86;hpb=3e4a8395a33db74da6ed7b99a561bf7e4d2b06b3)) and `AX_CHECK_LINK_FLAG` ([view diff](http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blobdiff;f=m4/ax_check_link_flag.m4;h=eb01a6ce135eca0fdde834b00c21b12202b09f3d;hp=db899ddd0859fdef4c168c1383cce7aacc26802e;hb=1a869696e4129279f7b99c3f9052717354b79a86;hpb=3e4a8395a33db74da6ed7b99a561bf7e4d2b06b3)) macros from Autoconf Archive. The previous version is functionally identical except that it was backwards-compatible with Autoconf 2.59 and later (by using `AS_IF` and `AS_VAR_GET` together in place of the newer `AS_VAR_IF`). I tested that this version of the macros still works properly with Autoconf 2.69 in Fedora 24.

sgalgano commented:
Thanks for the improvements.
